### ISSUE TITLE
[Bug] Clippy lint `use_self` false-positive

### ISSUE THREAD

**Issue by alexmozaidze** [NieDzejkob/enumflags2#40]
Putting `#![warn(clippy::use_self)]` at the top of `main.rs` and defining a bitflag enum gives a false-positive lint. It's nothing too important, but it would cool if it was fixed since `#[allow(clippy::use_self)]` doesn't stop the lint.
Minimal code to reproduce:
```rust
#![warn(clippy::use_self)]

use enumflags2::bitflags;

#[bitflags]
#[repr(u8)]
#[derive(PartialEq, Copy, Clone, Debug)]
#[allow(clippy::use_self)] // Doesn't stop the lint
enum MyEnum {
    MyVariant,
}

fn main() {
}
```
then run `cargo clippy` and *boom*, we have a false-positive.

**Comment by NieDzejkob** [comment]
Sure, it's not a problem to fix this. I'm currently trying to get this to trigger in the test suite, so I can make sure this doesn't regress later. Any ideas on how I can do that?

I added
```
#![deny(clippy::all, clippy::pedantic, clippy::use_self)]
```
to `test_suite/tests/bitflag_tests_2018.rs`, but `cargo +nightly clippy --all` and variations don't seem to run on the test suite...

**Comment by NieDzejkob** [comment]
Okay, looks like `--all-targets` is what I'm looking for.

**Comment by NieDzejkob** [comment]
The fix has now been published as enumflags2 v0.7.3.
